### PR TITLE
docs(blog): add shadcn post

### PR DESCRIPTION
**Pull request by necatiozmen**
I added shadcn post and example

### CODE REVIEW

File: examples/blog-refine-shadcn/src/pages/categories/create.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import { IResourceComponentsProps, useNavigation } from "@refinedev/core";
+import { useForm } from "@refinedev/react-hook-form";
+
+import { Form, FormItem, FormLabel, FormMessage } from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { Button } from "@/components/ui/button";
+
+import { ChevronLeft, ListIcon } from "lucide-react";
+
+export const CategoryCreate: React.FC<IResourceComponentsProps> = () => {
```
Comment (BatuhanW) [inline examples/blog-refine-shadcn/src/pages/categories/create.tsx:11]:
```suggestion
export const CategoryCreate: React.FC = () => {
```

File: examples/blog-refine-shadcn/src/pages/categories/create.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  const { list } = useNavigation();
+
+  const {
+    refineCore: { onFinish },
+    ...form
+  } = useForm({});
```
Comment (BatuhanW) [inline examples/blog-refine-shadcn/src/pages/categories/create.tsx:17]: We can pass generic type to useForm. 

`useForm<Category>(...)`

File: examples/blog-refine-shadcn/src/pages/categories/create.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            <FormItem>
+              <FormLabel className="m-2">Title</FormLabel>
+              <Input
+                type="text"
+                {...register("title", {
+                  required: "Title cannot be empty",
+                })}
+              />
+              <FormMessage className="text-destructive">
+                {(errors as any)?.title?.message as string}
```
Comment (BatuhanW) [inline examples/blog-refine-shadcn/src/pages/categories/create.tsx:58]: If we provide generic type, we shouldn't need `as any`.

File: examples/blog-refine-shadcn/src/pages/categories/edit.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+import { ChevronLeft, ListIcon } from "lucide-react";
+
+export const CategoryEdit: React.FC<IResourceComponentsProps> = () => {
+  const { list } = useNavigation();
+
+  const {
+    refineCore: { onFinish },
+    ...form
+  } = useForm({});
```
Comment (BatuhanW) [inline examples/blog-refine-shadcn/src/pages/categories/edit.tsx:17]:
```suggestion
  } = useForm();
```

File: examples/blog-refine-shadcn/src/pages/blog-posts/list.tsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+              </Button>
+            </div>
+          );
+        },
+      },
+    ],
+    [],
+  );
+
+  const { edit, show, create } = useNavigation();
```
Comment (BatuhanW) [inline examples/blog-refine-shadcn/src/pages/blog-posts/list.tsx:102]:
```suggestion
  const { create } = useNavigation();
```

**Review by omeraplak — APPROVED** [review]

**Pull request merged**